### ISSUE TITLE
performance issues

### ISSUE THREAD

**Issue by ff2000** [OfflineIMAP/offlineimap#317]
I currently have about 14000 Mails in my GMail INBOX. I only sync the INBOX folder. Syncing (without downloading any mails) takes ~1 minute. During that time I see high network traffic. I get ~240 kB/s, so (if my math is correct) every sync pulls ~14 MB. So 1kB for each message on the server.

Is that expected? Could there be done something to reduce network traffic?

My config:
```
[Account GmailAlp]
localrepository = GmailAlp-local
remoterepository = GmailAlp-remote
status_backend = sqlite
postsynchook = /home/franz/bin/tagmail.sh

[Repository GmailAlp-local]
type = Maildir
localfolders = ~/mail/GmailAlp
nametrans = lambda folder: {'archive': '[Google Mail]/Alle Nachrichten',
                            }.get(folder, folder)

[Repository GmailAlp-remote]
type = Gmail
sslcacertfile = /etc/ssl/certs/ca-certificates.crt
nametrans = lambda folder: {'[Google Mail]/Alle Nachrichten': 'archive',
                            }.get(folder, folder)
folderfilter = lambda folder: folder in ['INBOX'] # , 'Sent', '[Google Mail]/Gesendet']
idlefolders = ['INBOX']
```

**Comment by nicolas33** [comment]
OfflineIMAP is trying to get the changes for all the messages, including flag changes. I don't know how you got those numbers so I can't say.

I did a quick audit for a full sync vith v6.7.0 and I get ~1.18 MB traffic for approx 11000 mails: average: 0.1 kB/mail.
Emails are dispatched in different mailboxes, though.

Here are ways to improve the performance:

* reduce the number of mail in the INBOX (using archive folders)
* reduce the range of mails to sync with maxage
* run more quicksync and less full sync

**Comment by ff2000** [comment]
It was a rough measure from watching my network widget and running `time offlineimap`.
I now used wireshark to capture the traffic correctly. It are 5.45MB of total traffic. The time for a sync is now at 30 seconds, which is weird, because it took 1 minute at full network speed the last months. Probably a mistake on my side. But I think 5.5MB for each sync is still quite heavy.

I will give maxage a shot, now, though I remember there were some issues in the past. Can I use it safely without losing mail now?

I already use -q, but it only helps if there are no mails. I'm subscribed to some lists so I get new mails frequently.
Also archiving might help to a certain extent, but IMHO I would have to do it via webmail, because if I do it locally I simply move the traffic from syncing INBOX to syncing ARCHIVE.
Deleting isn't an option, I want to keep most of the mails around. I also want to keep them on the server as a backup so archiving locally and deleting on the server isn't an option, too).

**Comment by nicolas33** [comment]
> I will give maxage a shot, now, though I remember there were some issues in the past. Can I use it safely without losing mail now?

http://www.offlineimap.org/devel/2015/04/14/why-we-changed-maxage.html

> Also archiving might help to a certain extent, but IMHO I would have to do it via webmail, because if I do it locally I simply move the traffic from syncing INBOX to syncing ARCHIVE.

Not if you folderfilter the archive folders once synced.

> I also made a "small" mistake: I had the gmail-webinterface configured to show conversations, so I have 14000 threads. If I deactivate that feature I get ~57000 Messages. So I end up at about the same traffic for each message.

Yes, that's consistent which my measures.

**Comment by ff2000** [comment]
So syncing GMail<->Maildir is still OK with maxage. Thx!
I tried it and it seems to work.

**Comment by ff2000** [comment]
I ran ` offlineimap -a GmailAlp -k "Account GmailAlp:maxage=10"` for the last days. It worked. Just to make sure I ran ` offlineimap -a GmailAlp`. It didn't stop for more than 1 hour. There was a continious write of ~200kb/s for offlineimap (ran `itop`). After it finished I ran the same commands (first with maxage, than withou maxage) - and again I had quite some time (more than half an hour) offlineimap writing to disk with ~200kb/s.
Is that a known issue? I'm running latest offlineimap on Gentoo which is v6.5.7.
